### PR TITLE
Fix error on quantities when exist multiple organizations

### PR DESCRIPTION
**Pull request by Ganzt90**
Fixed error on quantities to replenish on Smart Browser Warehouse Replenish

### CODE REVIEW

**Comment by marcalwestf** [comment]
@Ganzt90  thnak you for your valuable contribution.

To help us review your changes easier and thus accept them earlier, it would be nice if you coud upload some screenshots before and after your changes. Otherwise we have to figure out the whole process.

Thanks!!

File: migration/393lts-394lts/7130_UpdateReplenishView.xml [hunk 1]
```diff
@@ -0,0 +1,445 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (yamelsenih) [inline migration/393lts-394lts/7130_UpdateReplenishView.xml:1]: Please change file name to 07130

**Review by yamelsenih — CHANGES_REQUESTED** [review]
Hi @Ganzt90 please fix it and add a step by step with a gif

**Comment by Ganzt90** [comment]
The following example shows what happens:

- Product: Bag50_Bag 50 Kg

Replenishment is configured, Maintain at maximum level in the HQ Warehouse, maximum level 40.00 minimum level 10.00.
The same configuration is done for the Store West warehouse.

image

image


When generating the Storage Detail reports it is observed that the Product has existence in the warehouse of the HQ organization and while it does not have existence in the warehouse of the Store West organization

image

Consulting the smartbrowser Warehouse Replenish for the HQ Warehouse and Store West warehouses, the following is obtained:

Error on SBReplenish

- The quantity to order in the HQ Warehouse of 12.
- The quantity to order in the Store West Warehouse of 12, where the correct thing for this warehouse should be 40 since the warehouse does not have the product in stock.

After correcting the SQL View of the Warehouse Replenishment Smartbrowser the following result is obtained:

Fixed Query on SBReplenish

**Comment by marcalwestf** [comment]
@Ganzt90  I loaded and executed the XML.
When I try to apply it, there are two errors:

> org.adempiere.exceptions.AdempiereException: Migration 7130 - Update Replenish View - 3.9.4 (ECA02) ; Step: 20; Type: SQL.  Application failed. 
> org.postgresql.util.PSQLException: Unterminated block comment started at position 0 in SQL /******************************************************************************
>  * Product: Adempiere ERP & CRM Smart Business Solution                       *
>  * This program is free software. Expected */ sequence
> 	at org.compiere.model.MMigration.apply(MMigration.java:178)
> 	at org.compiere.model.MMigration.apply(MMigration.java:176)

This error can be avoided by eliminating the java comments.

If the java comments are deleted, another error is thrown:

> Migration 7130 - Update Replenish View - 3.9.4 (ECA02) ; Step: 20; Type: SQL. Application failed. org.postgresql.util.PSQLException: ERROR: column s.m_warehouse_id does not exist
> 
> GROUP BY s.M_Product_ID) s ON(s.M_Product_ID = p.M_Product_ID AND s.M_Warehouse_ID=r_1.M_Warehouse_ID)
> Hint: Perhaps you meant to reference the column "r.m_warehouse_id". Position: 3207

Looking at the SQL, Relation "s" has no warehouse_id defined nor it is included in the GROPU BY clause; and also relation "r_1" doesn't exist.

Changing it as follows works, but I don't know if it is syntactically correct:
   

>  LEFT JOIN (SELECT s.M_Product_ID, 
>                 SUM(s.QtyOnHand) AS QtyOnHand, 
>                 SUM(s.QtyOrdered) AS QtyOrdered, 
>                 SUM(s.QtyReserved) AS QtyReserved, 
>                 SUM(s.QtyAvailable) AS QtyAvailable,
>                 M_Warehouse_ID 
>             FROM RV_Storage s
>             GROUP BY s.M_Product_ID, s.M_Warehouse_ID) s ON(s.M_Product_ID = p.M_Product_ID AND s.M_Warehouse_ID=r.M_Warehouse_ID)

**Comment by marcalwestf** [comment]
hi @Ganzt90 Germán, could you please modify the sequence no so it displays the same number as the XML file?
Selection_993

Thanks!

**Comment by marcalwestf** [comment]
running the _XML 07130_UpdateReplenishView.xml_, I get the following error:
Selection_994


... and looking at the SQL, the culprit is:
Selection_995

**Comment by marcalwestf** [comment]
@Ganzt90  the column _isActive_ is not defined for relation "r".
If you change instead "rep" for "r", the SQL works and the replenish report also.
Selection_996

**Review by marcalwestf — APPROVED** [review]
1. Checked, loaded, and executed XML file **from console** (with _RUN_MigrateXML.sh_).
2. View was modified accordingly.
3. Ran tests as described.
4. Error was corrected.
Test passed.

**Pull request merged**